src/pasta/fields/fq.rs

```diff
@@ -164,69 +164,61 @@ impl_binops_additive!(Fq, Fq);
 impl_binops_multiplicative!(Fq, Fq);
 
 /// INV = -(q^{-1} mod 2^64) mod 2^64
-const INV: u64 = 0x842cafd3ffffffff;
+const INV: u64 = 0x8c46eb20ffffffff;
 
 /// R = 2^256 mod q
 const R: Fq = Fq([
-    0x7379f083fffffffd,
-    0xf5601c89c3d86ba3,
+    0x5b2b3e9cfffffffd,
+    0x992c350be3420567,
     0xffffffffffffffff,
     0x3fffffffffffffff,
 ]);
 
 /// R^2 = 2^512 mod q
 const R2: Fq = Fq([
-    0x8595fa8000000010,
-    0x7e16a565c6895230,
-    0xf4c0e6fcb03aa0a2,
-    0xc8ad9106886013,
+    0xfc9678ff0000000f,
+    0x67bb433d891a16e3,
+    0x7fae231004ccf590,
+    0x96d41af7ccfdaa9,
 ]);
 
 /// R^3 = 2^768 mod q
 const R3: Fq = Fq([
-    0xa624f338075cdb5e,
-    0x57964eacb8fe21f2,
-    0xcb266d18c0413bc2,
-    0xa42cdf95f959577,
+    0x008b421c249dae4c,
+    0xe13bda50dba41326,
+    0x88fececb8e15cb63,
+    0x7dd97a06e6792c8,
 ]);
 
-/// `GENERATOR = 5 mod q` is a generator of the `q - 1` order multiplicative subgroup, and
-/// is also a quadratic non-residue.
+/// `GENERATOR = 5 mod q` is a generator of the `q - 1` order multiplicative
+/// subgroup, or in other words a primitive root of the field.
 const GENERATOR: Fq = Fq::from_raw([
     0x0000_0000_0000_0005,
     0x0000_0000_0000_0000,
     0x0000_0000_0000_0000,
     0x0000_0000_0000_0000,
 ]);
 
-const S: u32 = 34;
+const S: u32 = 32;
 
 /// GENERATOR^t where t * 2^s + 1 = q
 /// with t odd. In other words, this
 /// is a 2^s root of unity.
-///
-/// `GENERATOR = 5 mod q` is a generator
-/// of the q - 1 order multiplicative
-/// subgroup.
 const ROOT_OF_UNITY: Fq = Fq::from_raw([
-    0x1cbd3234869d57ec,
-    0xa287dd1b8084fbf,
-    0xf1dbcb645a987293,
-    0x113efc510dc03c0b,
+    0xa70e2c1102b6d05f,
+    0x9bb97ea3c106f049,
+    0x9e5c4dfd492ae26e,
+    0x2de6a9b8746d3f58,
 ]);
```

Comment:
I did not check this constant, but I am satisfied by its unit test.

Reply:
Checked up to here.